src/overlays/actors/ovl_En_Hg/z_en_hg.c

```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
+void func_80BCF7D8(EnHg* this, GlobalContext* globalCtx) {
+    if (this->actor.colChkInfo.health) {
+        if (this->collider.base.acFlags & AC_HIT) {
+            this->collider.base.acFlags &= ~AC_HIT;
+            func_80BCF68C(this);
+        }
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        if (this->actionFunc != func_80BCF6D0 && this->actionFunc != func_80BCF8A0 &&
+            this->actionFunc != func_80BCF95C) {
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF710.asm")
+void func_80BCF88C(EnHg* this) {
+    this->actionFunc = func_80BCF8A0;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF778.asm")
+void func_80BCF8A0(EnHg* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->cutscenes[this->unk218]) != 0) {
+        ActorCutscene_Start(this->cutscenes[this->unk218], &this->actor);
+        func_80BCF93C(this);
+    } else {
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+        }
+        ActorCutscene_SetIntentToPlay(this->cutscenes[this->unk218]);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF7D8.asm")
+void func_80BCF93C(EnHg* this) {
+    this->cutscenes[3] = 0x63;
+    this->cutscenes[2] = 0;
+    this->actionFunc = func_80BCF95C;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF88C.asm")
+void func_80BCF95C(EnHg* this, GlobalContext* globalCtx) {
+    if (func_800EE29C(globalCtx, 0x1E4U) != 0) {
+        u32 actionIndex = func_800EE200(globalCtx, 0x1E4U);
+        if (this->cutscenes[3] != globalCtx->csCtx.npcActions[actionIndex]->unk0) {
+            this->cutscenes[3] = globalCtx->csCtx.npcActions[actionIndex]->unk0;
+            switch (globalCtx->csCtx.npcActions[actionIndex]->unk0) {
+                case 1:
+                    this->currentAnimation = NULL;
+                    func_800BDC5C(&this->skelAnime, animations, 0);
+                    break;
+                case 2:
+                    this->cutscenes[2] = 0;
+                    this->currentAnimation = 3;
+                    func_800BDC5C(&this->skelAnime, animations, 3);
+                    break;
+                case 3:
+                    this->cutscenes[2] = 0;
+                    this->currentAnimation = 5;
+                    func_800BDC5C(&this->skelAnime, animations, 5);
+                    break;
+                case 4:
+                    this->cutscenes[2] = 0;
+                    this->currentAnimation = 7;
+                    if ((this->unk218 == 1) || (this->unk218 == 3)) {
+                        func_8019F128(0x3ABA);
```

Comment:
```suggestion
                        func_8019F128(NA_SE_EN_HALF_REDEAD_TRANS);
```
`func_8019F128` seems to be from an audio file, and the sfx matches up